skrub/_interpolation_join.py

```diff
@@ -0,0 +1,328 @@
+        ).set_index(original_index)
+
+    def fit_transform(self, left_table):
+        """Fit the estimators and perform the join.
+
+        Parameters
+        ----------
+        left_table : DataFrame
+            The table to transform.
+
+        Returns
+        -------
+        join : DataFrame
+            The result of the join between `left_table` and inferred rows from
+            ``self.right_table``.
+        """
+        return self.fit().transform(left_table)
+
+    def _get_estimator_assignments(self):
+        """Identify column groups to be predicted together and assign them an estimator.
+
+        In many cases, a single estimator cannot handle all the target columns.
+        This function groups columns that can be handled together and returns a
+        list of dictionaries, each with keys "columns" and "estimator".
+
+        Regression and classification targets are always handled separately.
+
+        Any column with missing values is handled separately from the rest,
+        because missing values in Y have to be dropped and the corresponding
+        rows may have valid values in the other columns.
+
+        When the estimator does not handle multi-output, an estimator is fitted
+        separately to each column.
+
+        """
+        right_table = self.right_table.drop(self._right_on, axis=1)
+        assignments = []
+        regression_columns = right_table.select_dtypes("number")
+        assignments.extend(
+            _get_assignments_for_estimator(regression_columns, self.regressor)
+        )
+        classification_columns = right_table.select_dtypes(
+            ["object", "string", "category"]
+        )
+        assignments.extend(
+            _get_assignments_for_estimator(classification_columns, self.classifier)
+        )
+        return assignments
+
+
+def _get_assignments_for_estimator(table, estimator):
+    """Get the groups of columns assigned to a single estimator.
+
+    (which is either the regressor or the classifier)."""
+    if not table.shape[1]:
+        return []
+    if not _handles_multioutput(estimator):
+        return [{"columns": [col], "estimator": estimator} for col in table.columns]
+    columns_with_nulls = table.columns[table.isnull().any()]
+    assignments = [
+        {"columns": [col], "estimator": estimator} for col in columns_with_nulls
+    ]
+    columns_without_nulls = list(set(table.columns).difference(columns_with_nulls))
+    if columns_without_nulls:
+        assignments.append({"columns": columns_without_nulls, "estimator": estimator})
+    return assignments
+
+
+def _handles_multioutput(estimator):
+    return getattr(estimator, "_get_tags")().get("multioutput", False)
+
+
+def _fit(X_values, right_table, target_columns, estimator):
+    estimator = base.clone(estimator)
+    kept_rows = right_table.loc[:, target_columns].notnull().all(axis=1).to_numpy()
+    X_values = X_values[kept_rows]
```

Comment:
here the X is a numpy array. the reason is that in case to_numpy() needs to make a copy (for example if the data is not contiguous in the pandas dataframe, or later if it is a polars dataframe), I prefer to do it only once in the parent process

Reply:
Oh right sorry I overlooked the `to_numpy()`